### PR TITLE
Migrate to Tuist 🌻 

### PR DESCRIPTION
**Pull request by jumaallan**
### **User description**
Story: https://app.shortcut.com/smileid/story/14654/ios-use-xcodegen-to-generate-xcodeproj-file

## Summary

Migrate to Tuist - Manages all necessary codegen operations so we can focus on writtng code and not fixing xcode issues everyday ):

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

___

### **PR Type**
Enhancement

___

### **Description**
• Migrate project build system from CocoaPods to Tuist
• Remove legacy build scripts and configuration files
• Update dependency management and project structure
• Temporarily disable Sentry integration during migration

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes</strong></td><td><details><summary>17 files</summary><table>
<tr>
  <td><strong>Project.swift</strong><dd><code>Add Tuist project configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-d9f98adc995a21cdce3b38f85bba1d56e2f0c54d977c4ef834235e93df3855d3">+44/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Tuist.swift</strong><dd><code>Add Tuist configuration file</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-409470ed2f5806676c6e7e9abb57d59f89fa0aa57c1b3c3fb1e52cb9d6c49bf0">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileID.swift</strong><dd><code>Define SmileID framework target configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-14349fbe7263736a3dadcb9e2db18bd197694169ab36b362b1ab99bf50a02dea">+24/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>run.sh</strong><dd><code>Add build script for Tuist workflow</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-d31ce0453051853c17ba2a5225b3d1bfab548e095bab0967d6acfd1b3ce1b35d">+10/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>HomeViewModel.swift</strong><dd><code>Temporarily disable Sentry SDK integration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-3f8b1a269c14969742c6799225df5be2fac7620413651d0e0e8695451d24c319">+6/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileID.podspec</strong><dd><code>Update Swift version requirement</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-7b77f15ea198e0bc969823c4cfbc250e2f35c075d092f421f4bffc86d00973a4">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Podfile</strong><dd><code>Update Podfile for new target structure</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-29b5a1cbec50129afcbbcdb6f92d1ba7d2691f19e2fa6716590dda60134413df">+3/-5</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Info.plist</strong><dd><code>Remove Main.storyboard reference</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-e0859d8096b078ab7c395aa60e431e0c7051a1900abbe101e0cdaa80f4583a47">+0/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Danger.swift</strong><dd><code>Remove Danger CI configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-b2dee3853e73e3ec2d4e3f083b5ed2abbf7e9dd41eb32141c1987477b7521ca6">+0/-32</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>update_compile_sources.rb</strong><dd><code>Remove legacy build script</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-4cb659630f5604a44eaa821d5dd3fa3e19ea468842c11eb157f2c33630880322">+0/-31</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>add_smile_config.rb</strong><dd><code>Remove legacy configuration script</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-b8a5a546f38c49de804e75a126466f157a1a43c5312504da8d9c028e6fdff942">+0/-25</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>build_frameworks.sh</strong><dd><code>Remove legacy framework build script</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-202d043ff3e39c75671b7d7aff9390647e915bd323b324d078b86284288b1d0a">+0/-32</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>action.yml</strong><dd><code>Remove GitHub Actions setup configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-9fb8259afcb532c9556e2d41ae1ee97cf6de54b486cef93a7ef1bf39433fa530">+0/-24</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>build-app.yml</strong><dd><code>Simplify CI workflow for Tuist</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-f02595ea0afb5cadd78ae828debf4e83e76ddc68642a1a4c7d9e1f8e61ab2192">+4/-56</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>carthage-release.yml</strong><dd><code>Remove Carthage release workflow</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-348007f37ff138b65dba466979581277dfcbf7f0567978a3d0293da0f3ff4af4">+0/-43</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>post_release.yml</strong><dd><code>Remove documentation generation workflow</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-7235ed47d43e54136a9c269dc833490c0d3b86f527f83edb114c879fa4ba6f66">+1/-33</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>sdk-release.yml</strong><dd><code>Remove SDK release workflow</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-704dfa813b791458c12d01ae0986baeed20c04c8c464932d90e1f9502239786c">+0/-22</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>Package.swift</strong><dd><code>Define external dependencies for Tuist</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-2c2060c16025934415d58ac106ea8f19a483b8111694da6904f60fddb91859dc">+28/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>EnhancedKycTest.swift</strong><dd><code>Add deprecated annotation to legacy test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-575a516cfe6441042bc6610c5c031b8b023b8088271098aaa10e5eb69e15e50a">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ConsentInformationTests.swift</strong><dd><code>Add deprecated annotations to legacy tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-73c9f6814a2396606aa851e3072c8a7e47bbe4f0ce989a81e3a073062c826397">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Additional files</strong></td><td><details><summary>16 files</summary><table>
<tr>
  <td><strong>SmileID.xcscheme</strong></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-ede60e859542b4760df83d123e2d732a8b4ddec0e592a0242e55136ba2c9c082">+0/-106</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>.arkana.yml</strong></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-3283f27a3379a9e7db0982b34e1a101fa5958b6ffe76aa6c86ee728b821d477b">+0/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>project.pbxproj</strong></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-88aa08ea238e57c3aa39fc93020b1e6185753c61d5413447ba6fb867b4cbe5fc">+0/-1088</a></td>

</tr>

<tr>
  <td><strong>SmileID-Example.xcscheme</strong></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-bf81db77608b9ebacb91e39806dbd7f3050f11d5d622a1a08d8688b945f28a88">+0/-112</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>SmileID_Tests.xcscheme</strong></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-e6f022a56343e1b2d064ca0e7d91e7ad37bd8de5b4dbdf8bf9f689ba2166fb9a">+0/-53</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Main.storyboard</strong></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-fa8d8b0cd2b89b3403cd19f9e183cf02822ec781c72a76580c1af02b37de36af">+0/-222</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>Appfile</strong></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-f9df02a2130921bfaef0a0076101225878e09bed5dde99053d0730d48f0df9c5">+0/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Fastfile</strong></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-fa18d481d482ad98725627850b5382ac6ccfdb574d803c0d9ebd9af6e574e0df">+0/-85</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Matchfile</strong></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-3678fbb8d325daef43acee906f55a73e8477d772f4ae5f8d2e366020d0d5e981">+0/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>README.md</strong></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-f7ea12e2c48c4e56fc7ba2470f208444dbca45e8d6c582e35dc6770e69d141c7">+0/-48</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Gemfile</strong></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-d09ea66f8227784ff4393d88a19836f321c915ae10031d16c93d67e6283ab55f">+0/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Rakefile</strong></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-ee98e028c59b193d58fde56ab4daf54d43c486ae674e63d50ddf300b07943e0f">+0/-128</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>project.pbxproj</strong></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-a85439f298e1ef5d8f4d6bfbe13b4de0b58d50f1cdb8869c00e5b1de71597530">+0/-9152</a></td>

</tr>

<tr>
  <td><strong>Package.resolved</strong></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-a7d4788aa112e64e20f70f58f7693dc841ecb0cc9552242812f7283ad4fd74a9">+0/-42</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileIDFramework.xcscheme</strong></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-5d2e47cf9948d9056b2492e70b3a45adea9c85f4f868194260ee1d9b9017e047">+0/-66</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Package.resolved</strong></td>
  <td><a href="https://github.com/smileidentity/ios/pull/344/files#diff-2b60ea18592f11ce008035d7f4e3aab58171ea4f2b3664826bc06f7e15ceddc1">+17/-8</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about PR-Agent usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

File: Tuist/ProjectDescriptionHelpers/SmileID.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import ProjectDescription
+
+public let SmileID = Target.target(
+    name: "SmileID",
+    destinations: .iOS,
+    product: .framework,
+    bundleId: "com.smileidentity.sdk",
+    deploymentTargets: .iOS("13.0"),
```
Comment (wangerekaharun) [inline Tuist/ProjectDescriptionHelpers/SmileID.swift:8]: Should this target the latest iOS versions, or does it not affect anything
